Implement a basic integration test

Geod24 opened the pull request.

Geod24 committed:
Implement a basic integration test

Geod24 committed:
Do not use Vibe.d logger from BlockStorage

Otherwise we have to link in Vibe.d in the integration test

Geod24 commented:
I am truly amazed how this just works.

AndrejMitrovic commented:
> I am truly amazed how this just works.

Put that on the Bosagora T-shirt.

AndrejMitrovic commented:
> Do not use Vibe.d logger from BlockStorage
> Otherwise we have to link in Vibe.d in the integration test

I don't understand this comment. Why can't we use vibe.d logging in integration tests?

Geod24 commented:
Now that #160 is merged this need a rebase

Geod24 commented:
> I don't understand this comment. Why can't we use vibe.d logging in integration tests?

Otherwise it attempts to link in Vibe.d's static ctors...

AndrejMitrovic commented:
Ah. And there's the reason why we don't like them. :)

Geod24 committed:
BlockStorage: Rename unMap to release

It seems that release was intended as the public facing API.
It is a more user-friendly name, and is used in the (versioned out) unittest.

Geod24 commented:
Updated, should be ready to go

AndrejMitrovic commented on source/agora/common/BlockStorage.d:
```diff
@@ -504,8 +502,7 @@ public class BlockStorage
         {
-            logError("saveIndex: %s",
-                () @trusted { return ex.message; }());
+            () @trusted { stderr.writeln("saveIndex: ", ex.message); }();
```
I assume we are going to replace this eventually with non-vibe.d logging?

Geod24 replied:
Yes

AndrejMitrovic commented on tests/unit/BlockStorage.d:
```diff
@@ -0,0 +1,134 @@
+{
+    const size_t count = 300;
+
+    string path = buildPath(getcwd, ".cache");
```
Hmm.. this doesn't make the test parallelizable yet though, since multiple tests will end up overwriting each others data.

But we can fix this later.

Geod24 replied:
Yep, at this point I wanted to just move (and execute) the test.
We should go with something similar to Ocean (sandbox), but let's add a few tests first.

AndrejMitrovic commented:
Good start. We could add filtering and parallel test runs later.
